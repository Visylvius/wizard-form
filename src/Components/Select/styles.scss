@import '../../variables';

.Select-label {
  color: $font-main;
  font-size: 12px;
}

.Select-mainWrapper {
  background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAA3klEQVRIS+3VMU9CMRTF8d8zBL+aizoQFhx0kUk33RzdYMNFXUFnYeGrYYyaJiUxJHDLSxodbNKpfeffc9/pbaPyaCrr+3OA++z4rtT5Pg5GuMnCY9yWQEoBE1xhlUUP8YDrCBIB0vojLvGO0yz4hm4JJAKcYYoPHGOZAUdYoIMBXrc5iQAHeMlzviFygj7O8dkWEJU4XI8chALRhn9AVKHf70VRTHu4wFfbmKZLNKt50dLBnna0imcMd/2I0phWa3Y/D1e1Xa9BCZJG0VuQNpaWKMx72xS1Fl5/WN3BN+AgJhnZQlq4AAAAAElFTkSuQmCC');
  background-position: calc(100% - .5rem),100% 0;
  background-size: 1.5em 1.5em;
  background-repeat: no-repeat;
  background-color: $plain-white;
  border-radius: 3px;
  box-sizing: border-box;
  cursor: pointer;
  line-height: 1.5em;
  padding: 0.5em 3.5em 0.5em 1em;
  margin-top: 5px;
  width: 100%;
  -ms-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
}