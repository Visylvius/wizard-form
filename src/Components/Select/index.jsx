import React from 'react';

import './styles.scss';

const Select = ({
  selectTestId,
  optionsArr,
  selectedVal,
  name,
  onChange,
  label,
}) => {
  return (
    <>
      <label className='Select-label'>{label}</label>
      <select
        data-testid={selectTestId}
        className='Select-mainWrapper'
        name={name}
        value={selectedVal}
        onChange={onChange}
      >
        <option value='default' defaultValue disabled>
          Choose...
        </option>
        {optionsArr.map((option, idx) => (
          <option
            data-testid={`SelectOption-${idx}`}
            key={option.value}
            value={option.value}
          >
            {option.displayValue}
          </option>
        ))}
      </select>
    </>
  );
};

export default Select;
