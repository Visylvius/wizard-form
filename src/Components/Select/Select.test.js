import React from 'react';
import renderer from 'react-test-renderer';

import Select from './index';

describe('render', () => {
  test('should render properly', () => {
    const tree = renderer.create(
      <Select
        optionsArr={[
          { displayValue: 'test', value: 'stuff' },
          { displayValue: 'default', value: 'default' },
        ]}
        selectedVal='default'
      />,
    );

    expect(tree).toMatchSnapshot();
  });
});
