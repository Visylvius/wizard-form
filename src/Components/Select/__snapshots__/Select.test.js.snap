// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render properly 1`] = `
Array [
  <label
    className="Select-label"
  />,
  <select
    className="Select-mainWrapper"
    value="default"
  >
    <option
      defaultValue={true}
      disabled={true}
      value="default"
    >
      Choose...
    </option>
    <option
      data-testid="SelectOption-0"
      value="stuff"
    >
      test
    </option>
    <option
      data-testid="SelectOption-1"
      value="default"
    >
      default
    </option>
  </select>,
]
`;
